clients/src/main/java/org/apache/kafka/clients/ClusterConnectionStates.java

```diff
@@ -105,11 +105,13 @@ public boolean isBlackedOut(String id, long now) {
     public long connectionDelay(String id, long now) {
         NodeConnectionState state = nodeState.get(id);
         if (state == null) return 0;
```

Comment:
The javadoc seems a bit misleading, this method doesn't handle slow/stalled connections for the connected case, does it?

Reply:
thanks for spotting

Reply:
The javadoc still seems confusing to me. It seems fine for `disconnected` state because we are returning remaining backoff. For `connecting` state, we are now returning setup timeout, isn't that to cover stalled state? And for `connected` state, we are relying on wakeup based on other events, so this method doesn't actually `handle slow/stalled connections`?

Reply:
right, updated to match the inline comment!